AppCenterIdentity/AppCenterIdentityTests/MSIdentityTests.m

```diff
@@ -67,59 +70,66 @@ - (void)testApplyEnabledStateWorks {
                                            appSecret:kMSTestAppSecret
                              transmissionTargetToken:nil
                                      fromApplication:YES];
-  MSServiceAbstract *service = (MSServiceAbstract *)[MSIdentity sharedInstance];
 
   // When
-  [service setEnabled:YES];
+  [self.sut setEnabled:YES];
 
   // Then
-  XCTAssertTrue([service isEnabled]);
+  XCTAssertTrue([self.sut isEnabled]);
 
   // When
-  [service setEnabled:NO];
+  [self.sut setEnabled:NO];
 
   // Then
-  XCTAssertFalse([service isEnabled]);
+  XCTAssertFalse([self.sut isEnabled]);
 
   // When
-  [service setEnabled:YES];
+  [self.sut setEnabled:YES];
 
   // Then
-  XCTAssertTrue([service isEnabled]);
+  XCTAssertTrue([self.sut isEnabled]);
 }
 
 - (void)testLoadAndDownloadOnEnabling {
 
   // If
-  MSIdentity *service = [MSIdentity sharedInstance];
   NSString *expectedETag = @"eTag";
   [self.settingsMock setObject:expectedETag forKey:kMSIdentityETagKey];
   NSData *serializedConfig = [NSJSONSerialization dataWithJSONObject:self.dummyConfigDic options:(NSJSONWritingOptions)0 error:nil];
-  OCMStub([self.utilityMock loadDataForPathComponent:[service identityConfigFilePath]]).andReturn(serializedConfig);
+  OCMStub([self.utilityMock loadDataForPathComponent:[self.sut identityConfigFilePath]]).andReturn(serializedConfig);
   OCMStub([self.ingestionMock sendAsync:OCMOCK_ANY eTag:OCMOCK_ANY completionHandler:OCMOCK_ANY]);
 
   // When
-  [service applyEnabledState:YES];
+  [self.sut applyEnabledState:YES];
 
   // Then
-  XCTAssertTrue([service.identityConfig isValid]);
+  XCTAssertTrue([self.sut.identityConfig isValid]);
   OCMVerify([self.ingestionMock sendAsync:nil eTag:expectedETag completionHandler:OCMOCK_ANY]);
 }
 
 - (void)testEnablingReadsAuthTokenFromKeychainAndSetsAuthContext {
 
   // If
-  MSIdentity *service = [MSIdentity sharedInstance];
   NSString *expectedToken = @"expected";
+  NSString *expectedHomeAccountId = @"fakeHomeAccountId";
+  MSALAccountId *mockAccountId = OCMPartialMock([MSALAccountId new]);
+  OCMStub(mockAccountId.identifier).andReturn(expectedHomeAccountId);
+  id accountMock = OCMPartialMock([MSALAccount new]);
+  OCMStub([accountMock homeAccountId]).andReturn(mockAccountId);
+  [self.settingsMock setObject:expectedHomeAccountId forKey:kMSIdentityMSALAccountHomeAccountKey];
   [MSMockKeychainUtil storeString:expectedToken forKey:kMSIdentityAuthTokenKey];
+  OCMStub([self.clientApplicationMock accountForHomeAccountId:expectedHomeAccountId error:[OCMArg setTo:nil]]).andReturn(accountMock);
+  MSIdentity *identityMock = OCMPartialMock(self.sut);
+  OCMStub([identityMock configAuthenticationClient]).andReturn(nil);
```

Comment:
Looks like the partial mock of MSIdentity is not necessary, and `[identityMock configAuthenticationClient]` is not needed. You can use `self.sut` instead  of the partial mock of MSIdentity, which is the pattern in this class.

Reply:
@annakocheshkova I reverted your change in `testEnablingReadsAuthTokenFromKeychainAndSetsAuthContext:` to keep it simple. As the name suggests, this test case only covers the scenario that MSAuthTokenContext authtoken is correctly set after enabling. IMO you don't need to cover home accountId because there're other tests covering that.